src/TreeViewList/TreeViewList.spec.tsx

```diff
@@ -294,3 +208,297 @@ test("should show no search results message when no search results are found", a
       .getByTestId("none-selected")
   ).toContainText("No search results.");
 });
+
+/**
+ * Helper function to get selected IDs from state.
+ *
+ * @param {FrameLocator} frame - The frame locator to interact with the Storybook iframe.
+ * @returns {Promise<string[]>} A promise that resolves to an array of selected IDs.
+ */
+const getSelectedIds = async (frame: FrameLocator): Promise<string[]> => {
+  return await frame
+    .locator("[data-selected-ids]")
+    .evaluate(el => JSON.parse(el.getAttribute("data-selected-ids") || "[]"));
+};
+
+// this test verifies that selecting the topmost element while it is in a collapsed state
+// correctly marks it as selected and returns all its leaf child IDs.
+test("should select the topmost element and return all leaf child IDs when collapsed", async ({
```

Comment:
```suggestion
test("should select the topmost element and return all enabled leaf child IDs when the element is in collapsed state and it gets selected", async ({
```
i guess I missed these last time, just made it clearer that we are not collapsing it, rather we click the element when it is in a collapsed state